### ISSUE TITLE
twisted/test/test_policies.py should use setUp and tearDown instead of setUpClass and tearDownClass

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2742]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2742|
|Type|task|
|Created|2007-07-20 14:34:26Z|

TimeoutTestCase and TestTimeout both use setUpClass to install a fake system clock and then uninstall it with a tearDownClass method.

This is dubious in the extreme.  The clock effectively represents process-global state.  trial really, really needs to specify exactly what the runtime context interactions between setUp, a test method, and tearDown are, and hopefully those interactions need to be such that installing a Clock in setUp and uninstalling it in tearDown aren't catastrophic for the test run. :)  (If this isn't clear, consider the case where a results object wants to schedule a delayed call - will the results object ever be invoked to do this after a single test method's setUp is run and before that method's tearDown is called?  the spec may need to talk about Deferred return values from test methods.)

In any case, there's almost certainly no guarantee, nor the possibility of supplying one, that trial won't need to may delayed calls across an entire TestCase full of test methods.

Both these classes should be changed to use setUp and tearDown now.

<details><summary>Searchable metadata</summary>

```
trac-id__2742 2742
type__task task
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1184942066000000 1184942066000000
changetime__1185658563000000 1185658563000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Done in setup-teardown-policies-2742

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Looks, good. Potential cleanups:
 * copyright
 * from future import
 * ```LoggingFactoryTestCase``` should reset ```TestLoggingFactory._counter``` to 0 in tearDown.
 * trailing whitespaces

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Did all of those except the ```_counter``` change.  Created [#2757](https://github.com/twisted/twisted/issues/2757) for that.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Great please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [20862]) Merge setup-teardown-policies-2742

Author: exarkun
Reviewer: therve
Fixes [#2742](https://github.com/twisted/twisted/issues/2742)

Replace setUpClass and tearDownClass on TestTimeout and TimeoutTestCase with
setUp and tearDown methods (provided by a mixin) which do the same thing.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
